### ISSUE TITLE
a11y: Fix error displays to ensure they contain 'role=alert'

### ISSUE THREAD

**Issue by nwmac** [rancher/dashboard#12812]
a11y Issue: 206

**Issue 206 - Error messages difficult to locate**

When users add invalid data in "User retention process schedule " input field, the mentioned error message gets dynamically displayed. However, screen reader users are not aware about the error message being displayed on the page.

Note, this occurs in several places that all need fixing:

Users Retention Settings
- "Invalid cron schedule

User: Create
- Passwords do not match 

Health Scale Component
Main Content
- Failed to scale test  down 
- Operation cannot be fulfilled on deployments.apps "test": the object has been modified; please apply your changes to the latest version and try again
(available after user selects 0 value in the progress bar) 

Example:

![image](https://github.com/user-attachments/assets/5b91101f-f235-43de-87b4-3b68314ee80c)

Suggested remeditation:

Specify role="alert" to ensure that the error message is announced to screen reader users.

**Comment by yonasberhe23** [comment]
Requires manual validation.

**Comment by a-arias** [comment]
**TESTS** 
Rancher : v2.11-d5a1481d04357fd04de2492ee1e7d4576ccca44c-head
UI: master 3665e77
Fix verified the reader can read the error messages, the error messages contain the `role="alert" ` ✅ 

<img width="1211" alt="Image" src="https://github.com/user-attachments/assets/38691614-5a2a-43c4-916a-2843c4fe0fef" />
